lib/rules/eqeqeq.js

```diff
@@ -71,7 +71,7 @@ module.exports = function(context) {
             }
 
             if (context.options[0] === "smart" && (isTypeOfBinary(node) ||
-                    areLiteralsAndSameType(node)) || isNullCheck(node)) {
+                    areLiteralsAndSameType(node) || isNullCheck(node))) {
```

Comment:
The grouping was wrong.

In fact this also was causing another bug, whereby `"smart"` did not accept `== null` as it should have.